### ISSUE TITLE
Ethereum Core Devs Meeting 46 Agenda

### ISSUE THREAD

**Issue by lrettig** [ethereum/pm#56]
# Ethereum Core Devs Meeting 46 Agenda
### [Constantinople Progress](https://github.com/ethereum/pm/issues/53)

# Agenda

1. Testing
2. Client Updates
3. Research Updates
4. [Constantinople](https://github.com/ethereum/pm/issues/53)
  a. Client progress.
  b. Ropsten block number.
  c. Can we do it before Devcon?

Time-allowing:

5. [EIP-1380: Reduced gas cost for call to self](https://github.com/ethereum/EIPs/pull/1380/files)
6. [EIP-1108: Reduce alt_bn128 precompile gas costs](https://eips.ethereum.org/EIPS/eip-1108)
7. [EIP-1057](https://eips.ethereum.org/EIPS/eip-1057) (ProgPoW) and [EIP-1355](https://eips.ethereum.org/EIPS/eip-1355) (Ethash1a)

**Comment by jacqueswww** [comment]
Hi There! I am eager to discuss reducing the gas fee on self-calls. EIP-1380 https://github.com/ethereum/EIPs/pull/1380/files.

**Comment by chfast** [comment]
I don't mind having it this week, but we agreed to have it one week later.

**Comment by 5chdn** [comment]
Please make it Sept 14 :+1:

**Comment by MoneroCrusher** [comment]
In my opinion issuance reduction & ASIC fork-off has to happen at the same time. With a issuance reduction only you are only accomodating 'investors' who have a very biased view and only want to make money by reducing inflation, it won't work anyway.
What is far more important is utility tokens & projects wanting to build their platform on the Ethereum network to have a stable underlying protocol without the risk of a centralized network and its implications.

Potential industries will only adopt or even consider their projects when they know the underlying protocol - Ethereum - is absolutely secure as a disruption (i.e. 51% attack, censoring of certain transactions, just imagine one industry trying to sabotage ERC20 token transaction of a competitor for a huge shipment in the shipping industry) could cause huge problems & uncertainty. The real & ultimate value in Ethereum lies in the utility tokens & smart contracts, not in reducing inflation from i.e. 7% to 4.5%. That's just gimmick & fairy dust and a desperate attempt from 'investors' to easen their bags.

Behold, miners are not less greedy (actually they are supposed to be greedy in this game theoretical protocol) but unlike 'investors' they fulfill a crucial function and bring actual value: security of the network.
So I'd advise you to reconsider your last stance of making a issuance-only without a ASIC-fork-off.

In the meantime ProgPOW is already live on a mainnet. Check out Bitcoin Interest (no endorsement). The block height for the fork was reached today and they're officially & successfully running ProgPOW. You can use that as a 'testnet' if you will & see that the algorithm is flawless and works as intended.
Now it's up to you to lead this project into a success and it will only happen if you uphold the core value of the protocol: smart contracts & decentralization (AKA security).

**Comment by sinabl** [comment]
I’m in full agreement with what Monerocrusher has written. In addition, centralization by miners will force the Eth team to ultimately cater more to their demands than the original neutral vision that It was founded upon. Let’s have a serious discussion on this topic in the meeting, There is absolutely nothing wrong with performing another fork in November if its decided in the meeting to move forward with ProgPOW

**Comment by MoneroCrusher** [comment]
Absolutely agree. I don't want to imagine the propaganda that will be thrown around by the big ASIC manufacturers when ETH finally switches to PoS.
Their hardware value & investments will go to zero, while GPU's value doesn't (it can be used for a lot of other stuff and also other coins).
Therefore you will have a smoother transition to PoS if you keep GPU miners in the game.

If you embrace ASICs now, miners will slowly adapt & get ASICs themselves and it will be even harder to get consensus and the community split will be more drastic.

It is absolutely critical to form an opinion about ASICs fast and clearly communicate what you will do about it. It's a big problen and I feel like the devs don't even fully realize it yet.

**Comment by Nmining** [comment]
I´m afraid there is already wavering in networks ,and miners are already switching coins. Eth value is getting very low(173usd) . https://ethermine.org/statistics. Moving to Progpow asap should at least give more hope for gpu mining in Eth.

**Comment by 5chdn** [comment]
Don't want to rush things, but regarding 4) Constantinople:

- Shall we discuss a block number for Ropsten? (Would be nice.)
- Is Constantinople still realistically happening before Devcon4? (Doubt that.)

**Comment by eosclassicteam** [comment]
@5chdn I think it is a right time to discuss a block number for ropsten network, most of the clients are now ready 👍

**Comment by MariusVanDerWijden** [comment]
I would like to talk about EIP-1057 (ProgPoW) and EIP-1355 (Ethash1a)
to give a timeline what we, the ethminer community, think what has to happen before we use them. Forking the consensus algorithm isn't as easy as Bitcoin Interest makes it seem. All miners, nodes and even light clients such as wallets have to be updated in order to accommodate the new algorithm. 

My main concern with ProgPoW is, that it was introduced by a company that specialize in providing special mining software. The open source community needs more time to analyse the characteristics of ProgPow before we can fork. Otherwise for profit companies could sell their mining software which outperforms the open source variants not by 3-5% (like now) but by way more (15+%). This would lead to a centralization of mining power again, controlled by the companies that provide the mining software. 

Therefore, we propose with Ethash1a  [EIP-1355](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1355.md) a minor change in the algorithm which would buy us more time to analyse ProgPoW.

**Comment by MoneroCrusher** [comment]
@MariusVanDerWijden
Supporting this. Buys time and allows for everyone to look into a permanent solution. I would even suggest making a dedicated HF for the permanent ASIC resistant PoW solution. Why wait for Istanbul HF if the solution is found after i.e. 2-3 months?
No need to risk the big masses adopting to ASICs because then you've crossed the event horizon and you can't go back without a community split.
Luckily right now ASICs are exotic and only found in huge farms and manufacturer's farms. But they will try to get them out there in the wild as fast as possible once they see that there is a serious interest in forking their ASICs off.

**Comment by mdaria510** [comment]
If issuance is reduced without something done about ASICs then it will be completely counter productive. There are no hobbyist ASIC miners. All ASIC miners will ever do is sell everything they mine, and they will sell quickly. Not dealing with this problem now is making a deal with the devil and making it more difficult to change things further down the line. 

Fact: Reducing issuance will put pressure on mining economics, such that the proportional of hash owned by ASICs *must* go up. At best you could argue that it might only go up a little bit, but it can only increase.

Supposedly the issuance reduction is to throw a bone to "investors" by cutting down on inflation. I know many "investors", and I have never met a single "investor" that said "the reason I'm selling ETH is because inflation is too high." No. The reason they're selling ETH is because the price is bleeding more and more, and they're losing faith. Average joe investor is very quickly shedding the belief that ETH is "the next bitcoin." 

Cutting the inflation will obviously prop up the price to some degree - in the long run. But you can't make a case for it turning around the current bleeding. The total effect is going to be a few % at best, the kind of swing we see on a daily basis. The net effect of that change will take months to years to work out. 

Cut off the GPU miners though, and you've dramatically increased selling pressure on day one after the fork because a much larger percent of mined ETH is going to be dumped by big centralized miners starting immediately after the fork, and it'll never stop. The block reward issuance will reduce the amount of ETH rewarded, but will not significantly reduce the amount of ETH sold by miners on the market. It might even increase it. There is a very real risk that it actually suppresses price in the short term. And at this stage, suppressing price in the short term is suppressing price in the long term.

At this early stage you should never lose sight of how immature the market is. "Investors" are not responding to fundamentals like inflation rate. "Investors" are not looking at supply. They're just not. Most don't even know what it is or where to find it. How else can you explain XLM and XRP having such huge valuations even though only a tiny fraction of the supply is even circulating, and the rest is held by centralized power? By fundamentals those tokens should be near worthless, but instead they're near the top of the pack. All "investors" are looking at is the price and the direction it's heading, and handing over mining to ASIC farms is going to increase selling pressure, suppress the price and drive new investors away. 

I can't take the "there's no time to do anything" argument seriously if there's actually a simple way to brick current ASICs and at least buy some time. If you ignore the concerns of GPU miners, not only would you be biting the hand that fed you over the past few years, you're trading a friendly idealistic bunch with a large percentage of holders and true believers for ruthless corporations that will not hesitate to bite back. If you dismiss this threat today, by the time you casually get around to dealing with it you'll have a much steeper hill to climb with much more resistance from organized opponents that have no interest in Ethereum whatsoever beyond mining and selling it. They will stir up an order of magnitude more chaos and divisiveness than a bunch of random GPU miners could ever hope to. If the price of ETH actually does pump over the next few years, that's going to be worth billions to Bitmain and friends. If the price dumps, mining returns will drop so low (especially with the BR decrease) that they'll end up owning more and more of it of the mining rewards. In either case, they'll have incentive to battle devs at every step, especially when they're an entrenched power. They're not going to roll over without a fight like most GPU miners. They will not be casually dismissed, and they will derail every and any plan developers have that goes against their interests. You need to cut the cancer out before it grows and spreads. That's why ASIC resistance was a core tenet of Ethereum from the start right? Assuming you can handle this at your leisure is not a safe assumption. Ignore it at your peril.

I get that punting ASIC resistance to the next fork is the easy thing to do, and reducing inflation is the easy thing to do to increase price. But the easy thing to do is rarely the right thing to do, there will be a price to pay down the line for taking the easy way out today. Every plan the dev team has is in jeopardy if you hand over running the network to people who have zero interest in seeing those plans come to fruition.

**Comment by NoBey** [comment]
@mdaria510   
The price of eth picks up again. The news of the production cut has changed the market, reducing the output of eth but not the revenue of gpu

**Comment by pdyraga** [comment]
Hey, would like to join the meeting to discuss  EIP-1108 on behalf of the Keep Network. We're happy to take lead on EIP updates and implementation across clients if an agreement can be reached.

**Comment by mdaria510** [comment]
@NoBey Wishful thinking is not a way to run a billion dollar blockchain.

**Comment by winsvega** [comment]
I am ready to start all tests review (which is done before every hardfork) 
This means all tests will be regenerated and a new version probably be added for most of the test cases (constantinople version) 

the test we are pending to add:

more sstore tests
extcodehash tests
transition tests around the fork point
difficulty change tests (not blokchain format)
fuzztests integration in form of state tests

**Comment by lrettig** [comment]
I added the requested topics to the agenda. Please note that we may not have time to get to all of them today; also, that I haven't been able to coordinate as closely with @Souptacular as usual given timezone differences and travel post-ETHBerlin, so I'll of course defer to his judgment regarding the agenda for today's call. @ytrezq my understanding of EIP-86 is that it was postponed due to the complexity of implementation and that it won't make it into Constantinople. I'll have to check the notes for more details on this but don't have time to do so before today's call.

**Comment by mhluongo** [comment]
I'd also be interested in a discussion of [EIP-1109](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1109.md) if it's not been covered in a previous call @jbaylina, as its relevant to precompile gas reduction

**Comment by holiman** [comment]
@ytrezq Epi 86 was considered, but eventually dropped, for a number of reasons. See the discusson the EIP PR for more context.

**Comment by MoneroCrusher** [comment]
As expected, ASIC resistance request from community is blatantly ignored and they think they can easily get away with it. This will backfire, and if done properly, I will join the backfire with my hashpower.

Would anyone here support a 0% fee ETH pool representing the community as a signal to the devs and that will collectively show support for an ASIC resistant implementation in Constantinople (ProgPOW, EIP-1355, or another ASIC-resistant solution)? I don't know how to run a pool properly (and to protect it from DDoS attacks), if anyone does this I will for sure support it.

**Comment by MoneroCrusher** [comment]
I was given the instructions on how to setup a testnet for ProgPOW with Geth client as a node and ethminer as a miner node, here they are.

```
## Instructions for setting up private Ethereum blockchain network

These set of instructions demonstrate setting up private Ethereum blockchain network with two nodes. One of the two nodes is a full client and other is a miner node. Go implementation of client,`geth` is used as client and `ethminer` is used as miner node. Latest version of `geth` is 1.8.15 and that of `ethminer` is 0.15.0. However, I encountered difficulties in setting up network using the latest versions. I had success with `geth` **1.8.13** and `ethminer` **0.13.0**. 

#### 1. Get `geth` and `ethminer` binaries

* `geth` binary can be downloaded from https://geth.ethereum.org/downloads/. Direct link for `geth` 1.8.13 on 64-bit Linux is https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.8.13-225171a4.tar.gz. Extract `geth` binary somewhere and add that location to `$PATH`.
* `ethminer` binary can be downloaded from https://github.com/ethereum-mining/ethminer/releases. Direct link for 0.13.0 on 64-bit Linux is https://github.com/ethereum-mining/ethminer/releases/download/v0.13.0/ethminer-0.13.0-Linux.tar.gz. Extract `ethminer` binary and add the location to `$PATH`.

#### 2. Client setup
1. Create a file `genesis.json` (the name can be changed) with following content

   ```
   {
     "config": {
       "chainId": 15,
       "homesteadBlock": 0,
       "eip155Block": 0,
       "eip158Block": 0
     },
     "alloc"      : {},
     "coinbase"   : "0x0000000000000000000000000000000000000000",
     "difficulty" : "0x20000",
     "extraData"  : "",
     "gasLimit"   : "0x2fefd8",
     "nonce"      : "0x0000000000000042",
     "mixhash"    : "0x0000000000000000000000000000000000000000000000000000000000000000",
     "parentHash" : "0x0000000000000000000000000000000000000000000000000000000000000000",
     "timestamp"  : "0x00"
   }
   ```

2. The client needs a data directory to store the blockchain and other stuff. It could be created anywhere, but I am assuming location of data directory is `~/eth-data/`

3. At least one account is needed for any kind of mining to happen. New account can be created with

   ```
   geth --datadir ~/eth-data/ account new
   ```
   This will prompt for password. Remember the password you use for this account.

4. Initialize the network with the genesis file created in step 1,

   ```
   geth --datadir ~/eth-data/ init genesis.json
   ```

5. Start the client

   ```
   geth --datadir ~/eth-data/ --networkid 15 --rpc --rpcport "1729" --rpccorsdomain localhost
   ```
   After executing this command, you should see a bunch of messages printed in console. Note down line that looks like this,
   ```
   IPC endpoint closed         endpoint=<path-to-data-dir>/geth.ipc
   ```
   Note down the full path for endpoint.

6. For reasons unclear to me, the mining does not work unless at least one block is mined on client itself. To do that, keep the client started in step 4 above running and open a second terminal. Execute following command

   ```
   geth attach <path-to-data-dir>/geth.ipc
   ```

   Use the endpoint path obtained in step 5. This will start a interactive console with the client. You can use this console to query useful information about network. The one which we might use often is `eth.blockNumber`. This gives the block number for last mined blocks. It should `0` right now. 
   Start miner in this console, using 

   ```
   > miner.start()
   ```
   Wait for few seconds. In the terminal 1 (client), you should see new blocks being added to the chain. They are indicated with messages like
   ```
   mined potential block                  number=25 hash=fa2d41…6b1511
   ```
   The number=... indicates the block number for last mined block. Now, you can stop the miner by going to terminal 2 (console) and executing,

   ```
   > miner.stop()
   ```

#### 3. Miner setup

1. At miner end, nothing much needs to be done. Open third terminal and launch miner with the localhost IP address and correct port as shown below

   ```
   ethminer -F http://127.0.0.1:1729
   ```

   You should see block number increasing in the client terminal in similar manner as in step 5.
   For newer versions of `ethminer`, use `-P` instead of `-F`.
```

**Comment by MoneroCrusher** [comment]
BREAKING NEWS
It will become worse and worse, **the "E9" is announced**.
https://www.coindesk.com/a-new-line-of-powerful-asic-miners-is-coming-to-ethereum/
Just as I have said, when they fear their investment becoming worthless, which an Ethereum ASIC-fork-off would do, they will try ANYTHING to get ASICs to the masses.

Do you believe an ASIC 7.37x more efficient than the E3 isn't long mining on the network?
If that doesn't make eth devs fork, we definitely know they have a hidden agenda and that [tinfoil hat on] maybe some of the devs are compromised by Bitmain. This is a **multi billion dollar market** after all. [tinfoil hat off]

**Comment by Nmining** [comment]
So.. This morning with a cup of coffee I´ve been walking around this few-gigahashes gpu farm as usual.  Steadily not a single drop on floor. And machines are doing well. But after reading this, I must look outside. So I let my coffee cup stay on table . Because I fear the worst. Through window, I can see the sun is shining... but not for long I know, more and more clouds will come and maybe some thunder. Unless someone push those ASICS away.

**Comment by OhGodAGirl** [comment]
> I would like to talk about EIP-1057 (ProgPoW) and EIP-1355 (Ethash1a)
> to give a timeline what we, the ethminer community, think what has to happen before we use them. Forking the consensus algorithm isn't as easy as Bitcoin Interest makes it seem. All miners, nodes and even light clients such as wallets have to be updated in order to accommodate the new algorithm.
> 
> My main concern with ProgPoW is, that it was introduced by a company that specialize in providing special mining software. The open source community needs more time to analyse the characteristics of ProgPow before we can fork. Otherwise for profit companies could sell their mining software which outperforms the open source variants not by 3-5% (like now) but by way more (15+%). This would lead to a centralization of mining power again, controlled by the companies that provide the mining software.
> 
> Therefore, we propose with Ethash1a [EIP-1355](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1355.md) a minor change in the algorithm which would buy us more time to analyse ProgPoW.

Good lord this again.

ProgPoW is open source. Anyone with a brain can verify what it does and does not do.

On top of that, no, it wasn't introduced by a company - it was a side project worked on by IfDefElse, of which I happen to be a member. This algorithm has _nothing_ to do with OhGodACompany or The Mineority Group. 

We've literally walked everyone through the code, and the code is nicely commented too on what it does and how it works.

Regardless, the E9 is out now. Here's a fun fact for you: it's not an ASIC. As in, it's going to be resistant to any of your tweaks. How, you ask? Well, that's a _ton_ of HBM2 in there. And those cores are going to be resistant in algorithm tweaks, because they can just be reprogrammed with some firmware. 

It's lightcache churn, to the max.

And guess what? There's an even better version coming. 

Enterprise miners will snap them all up, and control a majority of the hashpower, and when you have more than 20%, there are some _wonderful_ exploits you can do.

I'd like to make a point to the community: I would not throw my real identity in the mix if ProgPoW was not going to do what it says it can do, and if I would somehow personally profit. Have a think, for a long, hard moment, what that would _possibly_ accomplish, and how stupid that would possibly be when everyone and their mother knows who I am.

@Souptacular I would really like for misinformation and second guessing to stop happening, and I'd like to directly answer any questions the Parity and Ethminer teams have. If the timezones allow, we'll pull in Def and Else, too.

**Comment by MoneroCrusher** [comment]
@lrettig @vbuterin @5chdn @Souptacular It's so quiet around ASIC resistance talk. I think the community is no longer buying it. Come out and let us know what you think about hashrate centralization. And what you think about the new super ASIC that the company that announced it has full control over (that is 1 single entity).
Hypothetically: Would you guys be fine with me owning 80% of the Ethereum chain?

**Comment by MoneroCrusher** [comment]
Also, do you support the censorship on r/Ethereum? I made an anti-ASIC thread and it gained huge traction and was #1 on the frontpage quickly accelerating in upvotes.
First it was taken off the frontpage, then they even deleted my OP (https://web.archive.org/web/20180915123751/https://www.reddit.com/r/ethereum/comments/9fyldx/breaking_news_new_asic_with_737x_efficiency_of_e3/)
Now its back on, and I havent heard anything from a MOD why this has happened.
https://www.reddit.com/r/ethereum/comments/9fyldx/breaking_news_new_asic_with_737x_efficiency_of_e3/

Clearly you have a hidden agenda. Luckily you can't censor me on github. 👍 

I hope as many people as possible see this.
It's very strange how quiet that ASIC question by @Souptacular was in dev meeting 46. A little too quiet. Did some of the ETH devs have behind the doors meeting with ASIC manufacturers?

**Comment by benefit14snake** [comment]
This is crazy why are we being ignored?


What happens if they decide it’s more profitable to double spend on binance and bittrex instead of selling the miners to the public?

Please don’t ignore this security threat.

**Comment by MoneroCrusher** [comment]
@holiman This need to happen quicker. The algo is done, it only needs to be reviewed, tested in testnet and the clients need to be modified, for which @OhGodAGirl offered her services, she also said it's **easily** implementable by Const HF. The big chunk of work (mining algo) is already done and fully commented. If coins like BCI and Pirl can adapt it on short notices. How is it that Ethereum takes 11 months?

The network will already consist of 80% ASICs after Constantinople fork, forget about 2019, it will be fully owned by ASICs. By then you won't be able to make any forks anymore without huge community battles like in BTC, you absolutely underestimate this obvious threat. Once the manufacturer owns the chain, Ethereum takes a big hit in innovation & flexibility.
If not in Constantinople, at least make a complex enough tweak to the existing algo, so ASICs are kicked off and the behaviour is discouraged. maybe consult with @OhGodAGirl or @SChernykh or somebody else that can make good simple tweaks that will buy you a couple months. Then fork to ProgPOW as soon as it's been reviewed and tested, not in Istanbul. It will be too late by then.

With your signaling you're giving ASIC manufacturers hope and they will try to get everybody hooked on ASICs in the meantime. It's a one-way street.

**Comment by gfunksound** [comment]
I fully support ProgPow and EIP 1355 to be included in Constantinople. ASICS are the biggest threat to ETH right now and we need to address them ASAP. Let's not be owned by mining interests like Bitcoin

**Comment by ytrezq** [comment]
@holiman : where is that ᴘʀ discussion ?

**Comment by chfast** [comment]
Anyone is free to start integrating the ProgPoW with [ethminer](https://github.com/ethereum-mining/ethminer).

**Comment by MoneroCrusher** [comment]
@chfast that's long done? Since like April I think.
This is Bitcoin Interests ProgPOW github page that includes the finished ethminer:
https://github.com/BitcoinInterestOfficial/ProgPOW

**Comment by chfast** [comment]
**integrating**!

**Comment by holiman** [comment]
@ytrezq The discussion was unfortunately spread out a bit in various places, but @cdetrio summarized it pretty well here https://github.com/ethereum/EIPs/pull/208#issuecomment-313872489

**Comment by chfast** [comment]
The TODO list for ProgPoW adoption: https://ethereum-magicians.org/t/what-has-to-be-done-to-get-progpow-on-ethereum/1361.

**Comment by Nmining** [comment]
Btw just read news ethereum global hasrate gone down 20%

**Comment by 5chdn** [comment]
@Nmining We didn't do anything yet :)

**Comment by ytrezq** [comment]
@holiman : so why it wasn’t rescheduled on the Metropolis Part B fork coming next month as the linked comment suggest ?

**Comment by holiman** [comment]
> so why it wasn’t rescheduled on the Metropolis Part B fork coming 

@ytrezq I already answered:

> Epi 86 was considered, but eventually dropped, for a number of reasons. See the discusson the EIP PR for more context.

> as the linked comment suggest ?

Are you referring to what @lrettig said? 
 
> my understanding of EIP-86 is that it was postponed due to the complexity of implementation and that it won't make it into Constantinople. I'll have to check the notes for more details on this but don't have time to do so before today's call.

So he's wrong, it's not postponed, it was simply rejected. However, we're doing "Skinny Create2" instead, which is a second version of EIP 86/208 that is not as big/complex/quirky.

**Comment by MoneroCrusher** [comment]
@chfast @5chdn
So please be transparent to us, the community, about ASICs & Constantinople.
1. Is there a chance that ProgPOW will make it into Constantinople if reviewed in a good manner?
2. If you aswer 'no' to 1: is there a chance that EIP 1355 or another small change will make it into Constantinople to brick the current generation ASICs?

Please provide a thought out and definite answer. And please be aware that the community is in big support of forking ASICs off.
Thank you.

**Comment by 5chdn** [comment]
1. No
2. No

We have been working on Constantinople for 11 months now, and the hard-fork is final. Any other proposal, however small it is, has to go into the next hardfork.

**Comment by MoneroCrusher** [comment]
@5chdn
Too bad. IMO this is the end of decentralization of Ethereum. ASICs will now overtake the network hashrate completely over the next months, it'll be owned by around 3-4 companies until PoS and they will also have the power to influence all forks until PoS that goes against their interest, so good luck with that self-inflicted damage.

To the community that doesn't want to take this middle finger for years of dedication:
I don't know personally how to implement ProgPOW/rainforest, but if anyone does and also does the following things:

1. Fork current state of Ethereum
2. calls it Dethereum (DETH, Decentralized Ethereum)
3. Reduces DAG to 1 GB that allows older GPUs to participate too
4. Merges every innovation that the EF brings to light (including all changes in Constantinople & Istanbul that are not against decentralization)
5. Reduces issuance from 3 ETH to 1.75 ETH to undercut current Constantinople plans (2 ETH)
6. Either ProgPOW or Rainforest algorithm (both ASIC proof).

I will fully support it. ETH is dead for me if they're serious.

Good luck.

**Comment by benefit14snake** [comment]
Seriously. A lot of people are NOT happy with the broken promises. If you want to push your most staunch supporters away at a time where every centralized shitshow is claiming to be the next ETH killer - go ahead. Weird, I feel completely helpless as both a miner and a user so who really has say? Because it’s not the community. Maybe the initial ICO investors?

**Comment by Nmining** [comment]
Circumstances seems that there will be no reason for gpu miners to stay with eth. even bigger ones will suffer with new wave of asic. More bigger they are more they think about profits, and switch coins. theres goin to be only asic. Was that where everyone was aiming from start? white papers not so white anymore?

**Comment by eosclassicteam** [comment]
If ProgPOW didn't make a time with Constantinople we still have a chance to fork ethereum to anti-asic with different time schedule ( and I think this will help developers who are implementing Constantinople and help miners for implementing anti-asic algorithm )

**Comment by eosclassicteam** [comment]
Just think of previous forks without roadmap, there was DAO, EIP-150, EIP-155 and these hardforks were all patches for ethereum network, to fix vulnerability of it. 

If we think ASIC is a real threat to ethereum , why can't we fork later? With more development support??

**Comment by holiman** [comment]
I would agree, @eosclassicteam. It's not in Constantinople, but there's nothing preventing it from being implemented soon after -- if there's dev support, and it's considered a high enough threat/priority.

**Comment by mdaria510** [comment]
>           Seriously. A lot of people are NOT happy with the broken promises. If you want to push your most staunch supporters away at a time where every centralized shitshow is claiming to be the next ETH killer - go ahead. Weird, I feel completely helpless as both a miner and a user so who really has say? Because it’s not the community. Maybe the initial ICO investors?
> Edit: I watched Zcash hashrate fluctuate from 1.05gh to 1.61gh today. Which means one entity (likely Bitmain) could easily pull off a double spend off any major exchange.
> I know what you are thinking - that’s not half. Well they don’t need half if the other pools are split up. They just need the longest PoW - how much high than 50% dictates how far back in time you can reorganize.
> When these new ASICS come online there will be a time where no miner can “defend” against this security break since we won’t have to proper power to Claim longest POW.
> The exchanges stopped trading one time - will they do it again when several hundred million are stolen? Or will they just delist us?

You *shouldn't* feel helpless as a miner. You are empowered as a miner. Don't underestimate how much hash power GPU miners can collectively wield if someone were to effectively organize even a small percentage of them. We don't necessarily have to have the longest chain. We just have to have enough power to credibly back the existing chain and force exchanges to consider listing both chains. There needs to be enough of an uproar that press starts spreading the word that Ethereum is struggling to keep itself together. That alone will be enough to tank the price to counteract any supposed benefit from the issuance reduction. Think back to segwit2x - up until the cancellation there was enough theoretical support on both sides to force exchanges to consider the possibility that both would exist. No one ever had to mine a fork, they just had to believe that it plausibly could happen.  

@MoneroCrusher 
As soon as you fork and rename it you've lost. It's not an option. The only plausible way forward for GPU miners is to "go on strike" and commit to continuing to mine the existing chain and undermine the unity of Ethereum and force compromise. 

Because it's pretty clear that the devs consider us as second class citizens and not their most devoted users. It's time to remind them what decentralization actually looks like.

Devs don't control the blockchain. They issue proposals in the form of code, and users and particularly miners choose whether or not to approve those proposals by running that updated code. I do not approve of this proposal, and I will not run this code.

If there are pool operators out there that will publicly commit to continue running the existing chain past the fork, please step forward now and I will do everything in my power to spread the word. You have a good incentive to do so - it would make your pool unique and attract the attention of GPU miners worldwide, and you would therefore benefit from a significant increase in income prior to the fork, no matter what happens. I would do it myself if I wasn't prohibited in my state.

**Comment by Nmining** [comment]
From media https://cryptonewmedia.press/2018/09/18/how-much-did-asic-impact-crypto-mining-a-look-into-constantinople/

**Comment by MoneroCrusher** [comment]
@mdaria510 
That's a far greater idea than mine. Only when shit hits the fan we could think about it.
In your "resistance" scenario all that miners have to do is **nothing**. This is great.
We need to get some pools on our side because the pools will also get less revenue since ASICs & big farms will mine on their own private pools.
I for sure will not provide any hashing power to the new chain.

**Comment by 5chdn** [comment]
> This needs to be implemented shortly after Constantinople.

Then start working on it. I'm happy to accept any ASIC-resistant engine and tests for Parity Ethereum.

**Comment by MoneroCrusher** [comment]
@5chdn Would me getting a private testnet running help you with that? I got instructions on how to set one up with ProgPOW. I'm not a programmer though, but as far as I know OhGodAGirl is already in contact with you guys.

**Comment by gfunksound** [comment]
> I would agree, @eosclassicteam. It's not in Constantinople, but there's nothing preventing it from being implemented soon after -- if there's dev support, and it's considered a high enough threat/priority.

I think this is a very high threat against Ethereum, for the following reasons:

1. GPU miners are being pushed out of the market. This leads to higher levels of centralization with ASICs. Of course this is bad for one of the central pillars of blockchain, which is robust decentralization. 

2. We have many instances of ASICs completely dominating a coin and forcing them to make policy decisions at gun-point, and I don't think Ethereum core devs want to put themselves in this situation. 

3. The argument "ASICs aren't a threat because POS is 'around the corner' is a failed argument considering that by the time even the first implementation of POS rolls out, the E9 miner will already have shipped and be mining, and most GPUs will have dropped off the network. This makes transitioning to POS much harder since you'll be fighting all the ASICs with their break even timelines who won't want to switch. That will cause another divide in the community and result in an "Ethereum Classic Classic The Remix Part II The Sequel feat. The Zombie Doppelganger Twins" which is a disaster squared.  

ProgPow and EIP 1355 should be a priority because:

1. You literally have people like OhGodAGirl offering to do all the work, and the attitude is basically "we don't need your help". That seems very arrogant and is not in character with the open source community where everyone tries to help each other. 

2. The GPU miners are pissed that they are getting a block reduction down to 2, and not changing the POW to address ASICs is like a slap in the face. We need to get them back on-board, or else they will mine other coins just to spite ETH, and we are seeing that already with threats of this, and also spreading FUD, which brings me to my next point.

3. Endless sources of FUD surrounding this will happen and further tank the price. Ethereum cannot be successful at any stage if the ETH price is low. Investors will flee and interest will wane, and if there's only the hardcore developers there building "for the tech, man" you'll never get the holy grail of blockchain that everyone is supposedly building for: Mass Adoption

Thanks for reading

**Comment by holiman** [comment]
> You literally have people like OhGodAGirl offering to do all the work, and the attitude is basically "we don't need your help". That seems very arrogant and is not in character with the open source community where everyone tries to help each other.

@OhGodAGirl has been very helpful, and nobody (afaik) has said "we don't need your help". But it's a fact that switching consensus engine require quite a lot of work: 

> The TODO list for ProgPoW adoption: https://ethereum-magicians.org/t/what-has-to-be-done-to-get-progpow-on-ethereum/1361.

**Comment by HiddeHoogland** [comment]
Ethereum White Paper:

The Bitcoin mining algorithm works by having miners compute SHA256. This mining algorithm is vulnerable to two forms of centralization:

- First, the mining ecosystem has come to be dominated by ASICs (application-specific integrated circuits), computer chips designed for, and therefore thousands of times more efficient at, the specific task of Bitcoin mining. This means that Bitcoin mining is no longer a highly decentralized and egalitarian pursuit, requiring millions of dollars of capital to effectively participate in.

- Second, most Bitcoin miners do not actually perform block validation locally; instead, they rely on a centralized mining pool to provide the block headers. This problem is arguably worse: as of the time of this writing, the top three mining pools indirectly control roughly 50% of processing power in the Bitcoin network, although this is mitigated by the fact that miners can switch to other mining pools if a pool or coalition attempts a 51% attack.

This piece is inside of the original Ethereum Whitepaper, why aren't we pursuing the decentralization that we intended Ethereum to be?

Having 6000 running GPU's on the Ethereum network for a while, i never thought we would need this discussion. My hashrate will be supporting XMR for the time being.

**Comment by lrettig** [comment]
> it’s not like devs are in some double bind where there’s an immovable deadline and they’re being forced to do the impossible. The deadlines are self imposed. So they need to push the fork back if they have to and get it done.

This is not entirely true. The difficulty bomb is the immovable deadline. Call it an "excuse" if you like, but it requires us to hard fork relatively often, relatively on schedule. It's doing its job quite well.

**Comment by mdaria510** [comment]
It requires us to hard fork *if we want to keep the chain running*. But GPU miners are now being put in a position where there is effectively no difference between Constantinople making their hardware and investment obsolete or the difficulty bomb making their hardware and investment obsolete. I'd love nothing more than to see it removed because the only reason it exists in the first place was to get miners to act against their own self interest. Vitalik has said as much himself. Maybe they didn't forsee the perfect storm of a rally that brought in millions of miners only to have the price dump shortly thereafter, ultra efficient ASICs that made a joke of their ASIC resistance and being the only chain keeping those millions of GPUs afloat - but these are the consequences of the game theory inherent in that ploy.

We both know it's no big deal to remove it in a fork, so it is unequivocally a self-imposed deadline.

**Comment by MoneroCrusher** [comment]
Good news: https://twitter.com/nicksdjohnson/status/1042382028837203968
Nick Johnson now supports ProgPOW after chatting with @OhGodAGirl 
Hope my reddit thread helped accelerate that!

Timing is critical too, though. Devs knew about this since May. I think to restore miner community faith it should be done as fast as possible, not in Istanbul HF.
Why not in Istanbul HF? Because the network will punctually centralize with Constantinople HF at the end of October. Many GPUs will be turned off or turned to other coins and new hashrate growth will only come from ASICs.

So when Istanbul HF comes closer and they want to implement ProgPOW then, we will have even more of a community chaos, since some people will have switched to ASICs by then.

**Comment by 5chdn** [comment]
> Edit: I think it's worth pushing Constantinople back 1 month to be able to implement it, if this is deemed urgent.

My estimation is ~12 months. Instead of pushing back Constantinople for another year, let's just do another hardfork.

**Comment by MoneroCrusher** [comment]
@5chdn According to @OhGodAGirl 's last claim she said she could **easily** do it until Constantinople. What makes you think it's going to take 12 months? She has repeatedly said that the major part is already done. Only thing that's left is basically client work and that she would freely provide manpower for it.

**Comment by nootropicat** [comment]
@MoneroCrusher 
>Getting ASICs to the masses prevents Ethereum to fork because community doesn't want to when they own ASICs. This protects their investment.

If they manage to sell ASICs to the masses, ASIC mining is no longer centralized, making the decentralization argument moot.
Asic mining behaves as a physical proof of stake, offering much higher security than gpu mining (for the same block rewards).

**Comment by MoneroCrusher** [comment]
@nootropicat yeah unless they sell the ASIC that is 1:1 the efficiency of GPU to the masses for cheap and keep the one that is 8x more efficient to themselves, which is what they do.
Alternatively check out the Zcash problem or the Dash problem.

**Comment by holiman** [comment]
Well, if it's "ok" to push back Constantinople one month, how can it not be acceptible to roll it out in a HF one month after the non-pushed Constantinople? 

I would also support a switch to ProgPOW, but I do not think we should cram it into Constantinople. The implementation of the algorithm is _one_ aspect, but there are other things that need to be done to ensure a switch goes smoothly, and those things needs testing. 

My opinion is that we should consider a dedicated pow-switching HF shortly after Constantinople.

**Comment by nootropicat** [comment]
@MoneroCrusher there are at least two ethash asics providers now (Canaan and Bitmain) and they're competing, making that scenario unlikely and temporary at best. Keeping best products from the market invites new competition.
If you think it's a realistic long-term danger, why that's not the case in bitcoin?

>Alternatively check out the Zcash problem or the Dash problem.

That's too vague. What 'zcash problem'?

**Comment by mdaria510** [comment]
@holiman For various reasons that should be obvious enough that I don’t need to list them, I have very little faith in it actually being implemented in a timely manner (or at all) once the pressure is off.

**Comment by Nmining** [comment]
Grabs coffee cup from table. So what is going to happen to Etherium and stuff and is Ripple going to kick Etherium down to position 3 today? Oh!! #### this is cold...

**Comment by gfunksound** [comment]
@holiman I also agree about putting the transition to ProgPow in Constantinople. By reducing issuance to 2 ETH per block, suddenly many GPU miners are not profitable and will switch off the network. We've already seen a hashrate drop. This is the perfect opportunity for ASICs to sweep in and control the network. By delaying Constantinople to include ProgPow you don't risk anything except maybe a FudDesk article "Ethereum misses key hard fork deadline." or some garbage click-bait like that. Not a big deal. 

But not including it in Constantinople you risk ASICs owning the network, which opens the door to a very contentious HF "soon after" Constantinople if it happens at all before it is too late. Nobody wants Part II The Sequel of Ethereum Classic Classic.

**Comment by lrettig** [comment]
Closing in favor of #58
